zpy/objects.py

```diff
@@ -586,3 +586,83 @@ def restore_pose(
     obj = verify(obj)
     log.info(f"Restoring pose {pose_name} to object {obj.name}")
     obj.matrix_world = _SAVED_POSES[pose_name]
+
+
+def lighting_randomize(
+    path: Union[Path, str] = "hdri",
+    lighting_styles: List = ["bounded", "sun", "hdri"],
+    energy_jitter: bool = True,
+    lighting_styles_weights: List[int] = [3, 5, 2],
+) -> None:
+    """Randomizes Lighting Types.
+    Args:
+        path: str: Where to pull HDRI textures from
+        lighting_styles str: List of lighting styles to randomly choose from. Defaults to bounded, sun and hdri
+        energy_jitter: bool: Whether to jitter the lighting intensity for bounded and sun lights. Defaults to True
+        lighting_styles_weights: List[int]: List of 3 intergers corresponding to the lighting list that dictates how often each lighting style might appear.
+    """
+
+    if "bounded" in lighting_styles:
+        bounded_lights = []
+
+        for obj in bpy.data.lights:
+            if obj.type == "POINT" or obj.type == "SPOT" or obj.type == "AREA":
```

Comment:
This is clever, looking through the available lights in the scene and filtering based on types is a good idea.